plugins/browser-plugin-vimeo-tracking/src/types.ts

```diff
@@ -0,0 +1,115 @@
+import Player, { Error as VimeoError, QualityChangeEvent, TrackKind, VimeoCuePoint } from '@vimeo/player';
+import { InternalVimeoEvent, VimeoEvent } from './events';
+import { MediaEventType, MediaTrackingConfiguration } from '@snowplow/browser-plugin-media/src/types';
+
+/** The configuration object for the Vimeo tracking plugin */
+export interface VimeoTrackingConfiguration extends Omit<MediaTrackingConfiguration, 'captureEvents'> {
+  video: HTMLIFrameElement | Player;
+  captureEvents?: VimeoEvent[];
+}
+
+// An internal configuration type to avoid exposing `vimeoCaptureEvents` to the user
+export interface InternalVimeoTrackingConfiguration extends VimeoTrackingConfiguration {
+  /** Events filtered from `captureEvents` that are passed into the Vimeo player's `on` method */
+  vimeoCaptureEvents?: InternalVimeoEvent[];
+}
+
+/** A {@link Player} and its associated {@link VimeoTrackingConfiguration | configuration} */
+export type TrackedPlayer = {
+  player: Player;
+  eventListeners: Partial<Record<InternalVimeoEvent, (eventData: EventData | unknown) => void>>;
+  config: InternalVimeoTrackingConfiguration;
+};
+
+/** Associated type for the Vimeo "meta" schema */
+export type VimeoPlayerMetadata = {
```

Comment:
I didn't find where this type was used, is there an entity that is using it?

Reply:
I think I was going to use this as `SelfDescribingJson<VimeoPlayerMetadata>`, but then found out that wasn't possible. I've used it [here](https://github.com/snowplow/snowplow-javascript-tracker/pull/1212/commits/0d534e5ac54e50b7a30ce44f44125eda0b93ec7b#diff-a5acdcad1212365b743e3979aee311da7d3073cee339cd6d28a3043b1cb0003dR81) now.